src/c/c_client.cpp

```diff
@@ -30,517 +30,654 @@
-                         const char* device, const size_t device_length,
+void set_model_from_file(void *c_client,
+                         const char *key, const size_t key_length,
+                         const char *model_file, const size_t model_file_length,
+                         const char *backend, const size_t backend_length,
+                         const char *device, const size_t device_length,
                          const int batch_size, const int min_batch_size,
-                         const char* tag, const size_t tag_length,
-                         const char** inputs, const size_t* input_lengths,
+                         const char *tag, const size_t tag_length,
+                         const char **inputs, const size_t *input_lengths,
                          const size_t n_inputs,
-                         const char** outputs, const size_t* output_lengths,
+                         const char **outputs, const size_t *output_lengths,
                          const size_t n_outputs)
 {
-  /* This function sets a model stored in a binary file.
-  */
-  Client* s = (Client *)c_client;
-  std::string key_str = std::string(key, key_length);
-  std::string model_file_str = std::string(model_file, model_file_length);
-  std::string backend_str = std::string(backend, backend_length);
-  std::string device_str = std::string(device, device_length);
-  std::string tag_str = std::string(tag, tag_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == key || NULL == model_file || NULL == backend ||
+      NULL == device || NULL == tag || NULL == inputs || NULL == input_lengths ||
+      NULL == outputs || NULL == output_lengths)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string key_str(key, key_length);
+  std::string model_file_str(model_file, model_file_length);
+  std::string backend_str(backend, backend_length);
+  std::string device_str(device, device_length);
+  std::string tag_str(tag, tag_length);
 
   // Catch the case where an empty string was sent (default C++ client behavior)
   std::vector<std::string> input_vec;
-  if (n_inputs == 1 && input_lengths[0] == 0) {
-    input_vec = std::vector<std::string>();
-  }
-  else {
-    for(size_t i=0; i<n_inputs; i++) {
-      input_vec.push_back(std::string(inputs[i], input_lengths[i]));
+  if (1 != n_inputs || 0 != input_lengths[0]) {
+    for (size_t i=0; i<n_inputs; i++) {
+      input_vec.push_back((NULL != inputs[i] && 0 != input_lengths[i]) ?
+			  std::string(inputs[i], input_lengths[i]) :
+			  std::string(""));
     }
   }
 
   std::vector<std::string> output_vec;
-  if (n_outputs == 1 && output_lengths[0] == 0) {
-    output_vec = std::vector<std::string>();
-  }
-  else {
-    for(size_t i=0; i<n_outputs; i++) {
-      output_vec.push_back(std::string(outputs[i], output_lengths[i]));
+  if (1 != n_outputs || 0 != output_lengths[0]) {
```

Comment:
I know the advantage of the rvalue on the left side of ``==`` comparison helps guard against accidental assignment, but the isocpp core guidelines seem to just use the ``n_outputs != 1`` for readability.  We should talk as a group about this so that we can be consistent.  At the very least, it seems hard to switch between the two coding styles.

Reply:
Agree we should have a group discussion on this

Reply:
Per group discussion, tabling use of Yoda conditionals for now